src/neural/cuda/layers.cc

```diff
@@ -836,24 +837,29 @@ void FusedWinogradConvSELayer<DataType>::Eval(
   DataType* transformed_output =
       transformed_input + scratch_size / (2 * sizeof(DataType));
 
-  InputTransform<DataType>(N, c_input_, transformed_input, input);
+  InputTransform<DataType, false>(N, c_input_, transformed_input, input);
 
   cublasRowMajorMatrixMul(transformed_input, transformed_weights_, transformed_output, N*4, C, c_input_, 36, cublas);  
 
   if (has_se_ && use_relu_ && use_bias_ && skip_add_)
-    OutputTransform<DataType, true, true, true, true>(
+    OutputTransform<DataType, true, true, true, true, false, false>(
         N, C, se_k_, output, transformed_output, input2, biases_, w1_, b1_, w2_,
         b2_);
-  else if (!has_se_ && use_relu_ && use_bias_ && !skip_add_)
-    OutputTransform<DataType, false, true, true, false>(
-        N, C, 0, output, transformed_output, nullptr, biases_, nullptr, nullptr,
-        nullptr, nullptr);
-  else if (!has_se_ && use_relu_ && use_bias_ && skip_add_)
-    OutputTransform<DataType, false, true, true, true>(
+  else if (!has_se_ && use_relu_ && use_bias_ && !skip_add_) {
+    if (op_nhcw_)
+      OutputTransform<DataType, false, true, true, false, false, true>(
+          N, C, 0, output, transformed_output, nullptr, biases_, nullptr,
+          nullptr, nullptr, nullptr);
+    else
+      OutputTransform<DataType, false, true, true, false, false, false>(
```

Comment:
This looks like it can be simplified to `OutputTransform<DataType, false, true, true, false, false, op_nhcw_>(`

Reply:
The problem with this (and all other template params) is that they need to be compile time constants, so passing the op_nhcw_ variable as template param doesn't work. I think it would have been lot cleaner if we made the class itself a template on all those variables (has_se, use_relu_, use_bias_, skip_add_ and op_nhcw_).